File: src/main/java/io/github/mgrablo/BiblioNode/mapper/BookMapper.java
package io.github.mgrablo.BiblioNode.mapper;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import io.github.mgrablo.BiblioNode.dto.BookRequest;
import io.github.mgrablo.BiblioNode.dto.BookResponse;
import io.github.mgrablo.BiblioNode.model.Book;

@Mapper(componentModel = "spring")
public interface BookMapper {
	@Mapping(target = "id", ignore = true)
	@Mapping(target = "author", ignore = true)
	@Mapping(target = "createdAt", ignore = true)
	@Mapping(target = "modifiedAt", ignore = true)
	@Mapping(target = "available", ignore = true)
	Book toEntity(BookRequest bookRequest);

	@Mapping(source = "author.id", target = "authorId")
	@Mapping(source = "author.name", target = "authorName")
	BookResponse toResponse(Book book);
}

File: biblionode-ui/src/app/features/books/domain/mappers/book.mapper.ts
import { BookResponse } from '../../data-access/dto/book-response';
import { Book } from '../models/book.model';

export class BookMapper {
  static toBook(response: BookResponse): Book {
    console.log('Mapping BookResponse to Book:', response);
    return {
      id: response.id,
      title: response.title,
      isbn: response.isbn,
      authorName: response.authorName,
      availableCopies: response.available ? 1 : 0,
      coverUrl: response.coverUrl,
      description: response.description,
    };
  }

  static toBookList(responses: BookResponse[]): Book[] {
    return responses.map(BookMapper.toBook);
  }
}
